src/AppBundle/Entity/TonMacronFriendInvitation.php

```diff
@@ -0,0 +1,157 @@
+
+    /**
+     * @ORM\Column(type="smallint", length=3, options={"unsigned": true})
+     */
+    private $friendAge;
+
+    /**
+     * @ORM\Column(length=6)
+     */
+    private $friendGender;
+
+    /**
+     * @ORM\Column(length=50)
+     */
+    private $friendPosition;
+
+    /**
+     * @ORM\Column(nullable=true)
+     */
+    private $friendEmailAddress;
+
+    /**
+     * @ORM\Column(length=50, nullable=true)
+     */
+    private $authorFirstName;
+
+    /**
+     * @ORM\Column(length=50, nullable=true)
+     */
+    private $authorLastName;
+
+    /**
+     * @ORM\Column(nullable=true)
+     */
+    private $authorEmailAddress;
+
+    /**
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $mailSubject;
+
+    /**
+     * @ORM\Column(type="text", nullable=true)
+     */
+    private $mailBody;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $createdAt;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="AppBundle\Entity\TonMacronChoice", fetch="EAGER")
+     * @ORM\JoinTable(
+     *   name="ton_macron_friend_invitation_has_choices",
+     *   joinColumns={
+     *     @ORM\JoinColumn(name="invitation_id", referencedColumnName="id")
+     *   },
+     *   inverseJoinColumns={
+     *     @ORM\JoinColumn(name="choice_id", referencedColumnName="id")
+     *   }
+     * )
+     */
+    private $choices;
+
+    public function __construct(UuidInterface $uuid, string $friendFirstName, string $friendAge, string $friendGender, string $createdAt = 'now')
+    {
+        $this->uuid = $uuid;
+        $this->friendFirstName = $friendFirstName;
+        $this->friendAge = $friendAge;
+        $this->friendGender = $friendGender;
+        $this->choices = new ArrayCollection();
+        $this->createdAt = new \DateTimeImmutable($createdAt);
+    }
+
+    public function getFriendFirstName(): string
+    {
+        return $this->friendFirstName;
+    }
+
+    public function getFriendAge(): int
+    {
+        return $this->friendAge;
+    }
+
+    public function getFriendGender(): string
+    {
+        return $this->friendGender;
+    }
+
+    public function getFriendPosition(): ?string
+    {
+        return $this->friendPosition;
+    }
+
+    public function getFriendEmailAddress(): ?string
+    {
+        return $this->friendEmailAddress;
+    }
+
+    public function getAuthorFirstName(): ?string
+    {
+        return $this->authorFirstName;
+    }
+
+    public function getAuthorLastName(): ?string
+    {
+        return $this->authorLastName;
+    }
+
+    public function getAuthorEmailAddress(): ?string
+    {
+        return $this->authorEmailAddress;
+    }
+
+    public function getMailSubject(): ?string
+    {
+        return $this->mailSubject;
+    }
+
+    public function getMailBody(): ?string
+    {
+        return $this->mailBody;
+    }
+
+    public function getCreatedAt(): \DateTimeImmutable
+    {
+        if ($this->createdAt instanceof \DateTimeImmutable) {
```

Comment:
This will never be true after Doctrine loaded the entity, instead you can do:
```php
if (!$this->createdAt instanceof \DateTimeImmutable) {
    $this->createdAt = \DateTimeImmutable::createFromMutable($this->createdAt);
}

return $this->createdAt;
```